### DISCUSSION TITLE
Can't get REPL running

### DISCUSSION THREAD

**Discussion by radlyeel** [micropython/micropython#10679] [ESP8266]
I just received my pack of ESP 12F (ESP8266) development boards a few hours ago, and I'm stumped on getting the MicroPython REPL working.

Symptoms: After "esptool.py erase_flash" and "esptool.py  --baud 460800 --port /dev/cu.usbserial-0001 write_flash 0 downloads/esp8266-20190125-v1.10.bin" (both of which seemed to work) one of the LEDs flashes at a high rate continually (all my life that has meant data activity on a UART).  Then when I run "screen /dev/cu.usbserial-0001 115200", the terminal window fills up with randomness, which I've always associated with mismatched baud rates or other UART settings.

Environment:  MacOs 11.7.2, SiLabs drivers 6.0.2 freshly installed today.  in addition to the v1.10 shown above, I get the same behavior with v1.19.1, both 1m and 2m sizes.  Any correction would be appreciated.

**Comment by davefes** [comment]
When I was playing with these parts more than a year ago this is a more typical upload command:
```py
esptool.py --port /dev/ttyS3 --baud 115200 write_flash --flash_size=1MB -fm dout 0x0 the_image_file.bin
```
I recall there are 4 different options for `dout`

**Reply by robert-hh** [reply]
It would suggest trying the various `-fm` options as well, with `dout` first. But the --flash_size option has to match the board's flash size, and for  a 1M deviec the 1M fiurmware images have to be used. 
b.t.w.: There is not need to use a 2019 firmware version. You can get a recent firmware at https://micropython.org/download/?port=esp8266, including the nightly builds.

**Reply by radlyeel** [reply]
There seems to be progress.  This command left the board in a "quiet" state.  

esptool.py  --baud 115200 --port /dev/cu.usbserial-0001 write_flash -fm dio -fs=detect 0 downloads/esp8266-2m-20220618-v1.19.1.bin

"quiet" in the sense that the LEDis no longer flashing continuously.  But I can't connect to the serial port with "screen /dev/cu.usbserial-0001" or "rshell -p /dev/cu.usberial-0001".  They both report that the port is busy.

**Reply by robert-hh** [reply]
That could be a access right problem. I do not know Mac well, but if it supports `sudo`, try calling screen etc. with sudo.
`sudo screen /dev/cu.usbserial-0001`.

**Reply by radlyeel** [reply]
It seems that "screen" is not well- behaved.  To exit, one presses ^A, then D.  that exits without closing the connection.  There is a cumbersome pair of commands to clean up, which I'll make a script for. But first, now that I know it's working, I'll check out  rshell.

**Reply by robert-hh** [reply]
The `official` tool maintained by MicroPython is mpremote. When started without options, it gives you a terminal for the REPL prompt. mpremote can be installed with pip.
Edit: Screen drove me crazy when I tried it once (and never again).
